app/test_main.py

```diff
@@ -1 +1,28 @@
-# write your code here
+import pytest
+
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, connection, expected", [
+        (True, True, "Accessible"),
+        (True, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (False, False, "Not accessible"),
+    ]
+)
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+def test_can_access_google_page(
+        mock_valid_url: callable,
```

Comment:
Indicate in the type annotation that you expect to obtain a `Mock` for `mock_valid_url` and `mock_connect`.